devtools/klap-decrypt.py

```diff
@@ -0,0 +1,272 @@
+        if hasattr(
+            packet, "http"
+        ):  # this is redundant, as pyshark is set to only load http packets
+            if hasattr(packet.http, "request_uri_path"):
+                uri = packet.http.get("request_uri_path")
+            elif hasattr(packet.http, "request_uri"):
+                uri = packet.http.get("request_uri")
+            else:
+                uri = None
+            if hasattr(packet.http, "request_uri_query"):
+                query = packet.http.get("request_uri_query")
+                # use regex to get: seq=(\d+)
+                seq = re.search(r"seq=(\d+)", query)
+                if seq is not None:
+                    operator.seq = int(
+                        seq.group(1)
+                    )  # grab the sequence number from the query
+            data = packet.http.file_data if hasattr(packet.http, "file_data") else None
+            match uri:
+                case "/app/request":
+                    if packet.ip.dst != device_ip:
+                        continue
+                    message = bytes.fromhex(data.replace(":", ""))
+                    try:
+                        plaintext = operator.decrypt(message)
+                        myDict = json.loads(plaintext)
+                        print(json.dumps(myDict, indent=2))
+                        finalArray.append(myDict)
+                    except ValueError:
+                        print("Insufficient data to decrypt thus far")
+
+                case "/app/handshake1":
+                    if packet.ip.dst != device_ip:
+                        continue
+                    message = bytes.fromhex(data.replace(":", ""))
+                    operator.local_seed = message
+                    response = None
+                    while (
+                        True
+                    ):  # we are going to now look for the response to this request
+                        response = capture.next()
+                        if (
+                            hasattr(response, "http")
+                            and hasattr(response.http, "response_for_uri")
+                            and (
+                                response.http.response_for_uri
+                                == packet.http.request_full_uri
+                            )
+                        ):
+                            break
+                    data = response.http.file_data
+                    message = bytes.fromhex(data.replace(":", ""))
+                    operator.remote_seed = message[0:16]
+                    operator.remote_auth_hash = message[16:]
+
+                case "/app/handshake2":
+                    continue  # we don't care about this
+                case _:
+                    continue
+    except StopIteration:
+        break
+
+# save the final array to a file
+with open(output_json_name, "w") as f:
+    f.write(json.dumps(finalArray, indent=2))
+    f.write("\n" * 1)
+    f.close()
```

Comment:
The command could take `-o` (and `--output`) as an option, and default to printing out the results in the stdout.

Reply:
Done in fd2dfa12a8552623cf4bba86be6ddef1d5c8fe28